src/item.cpp

```diff
@@ -1947,12 +1947,18 @@ std::string item::info( std::vector<iteminfo> &info, const iteminfo_query *parts
     } ) ) {
 
         info.emplace_back( "QUALITIES", "", _( "Contains items with qualities:" ) );
-
+        std::map<quality_id, int> most_quality;
         for( const auto &e : contents ) {
             for( const auto &q : e.type->qualities ) {
-                name_quality( q );
+                most_quality.emplace( q );
```

Comment:
emplace returns a std::pair<iterator, bool>, which can be used like:
```
auto emplace_result = most_quality.emplace( q );
if( !emplace_result.second && emplace_result.second->second < q.second ) {
    emplace_result.second->second = q.second;
}
```
I don't really care about the repeat lookups, but I don't like the ability for at() to throw an exception.

Reply:
correct me if i'm wrong, but emplace will insert the key and value if they don't exist currently in the map, which means at() will have a key to find whether emplace returns true or false

Reply:
The interface still specifies that it throws an exception, and we dont generally handle exceptions.
If youre sure it will not fail you should use the [] operator.

Reply:
changed to use the emplace result